.github/workflows/build-and-test.yaml

```
@@ -11,7 +11,21 @@ jobs:
+      - name: Install Poetry
+        uses: snok/install-poetry@v1
+        with:
+          virtualenvs-create: true
+          virtualenvs-in-project: true
+          virtualenvs-path: .venv
+          installer-parallel: true
+      - name: Install dependencies
+        run: poetry install --no-interaction --no-root
       - uses: pre-commit/action@v3.0.1
+      - name: Run if any of the listed files above is changed
+        if: steps.changed-py-files.outputs.any_changed == 'true'
```

actionlint (1.7.7)

property "changed-py-files" is not defined in object type {}

(expression)